plugins/inputs/nftables/nftables.go

```diff
@@ -0,0 +1,185 @@
+`
+
+type Nftable struct {
+	Metainfo          *Metainfo
+	Rules             []*Rule
+	JSONSchemaVersion int `json:"json_schema_version"`
+}
+
+// UnmarshalJSON handles custom parsing of the nftable output which is
+// designed in a generic that is not compatible with the generic parser.
+func (nftable *Nftable) UnmarshalJSON(b []byte) error {
+	var atable map[string][]map[string]json.RawMessage
+	if err := json.Unmarshal(b, &atable); err != nil {
+		return fmt.Errorf("unable to unmarshal: %s", b)
+	}
+	// []map[string]interface
+	nfthings := atable["nftables"]
+	for _, nfthing := range nfthings {
+		hasKey := func(key string) bool { _, ok := nfthing[key]; return ok }
+		switch {
+		case hasKey("metainfo"):
+			var mi Metainfo
+			err := json.Unmarshal(nfthing["metainfo"], &mi)
+			if err != nil {
+				return fmt.Errorf("unable to parse metadata: %w", err)
+			}
+			nftable.Metainfo = &mi
+		case hasKey("rule"):
+			var r Rule
+			err := json.Unmarshal(nfthing["rule"], &r)
+			if err != nil {
+				return fmt.Errorf("unable to parse rule: %w", err)
+			}
+			nftable.Rules = append(nftable.Rules, &r)
+		default:
+			// something we are not parsing
+		}
+	}
+	return nil
+}
+
+type Metainfo struct {
+	Version           string `json:"version"`
+	ReleaseName       string `json:"release_name"`
+	JSONSchemaVersion int    `json:"json_schema_version"`
+}
+
+type Rule struct {
+	Family  string
+	Table   string
+	Chain   string
+	Comment string
+	Counter *Counter
+}
+
+// UnmarshalJSON handles properly extracing the counter expression from
+// the Exprs array input
+func (rule *Rule) UnmarshalJSON(b []byte) error {
+	var raw struct {
+		Family  string                       `json:"family"`
+		Table   string                       `json:"table"`
+		Chain   string                       `json:"chain"`
+		Comment string                       `json:"comment"`
+		Exprs   []map[string]json.RawMessage `json:"expr"`
+	}
+	if err := json.Unmarshal(b, &raw); err != nil {
+		return fmt.Errorf("unable to unmarshal: %s", b)
+	}
+	rule.Family = raw.Family
+	rule.Table = raw.Table
+	rule.Chain = raw.Chain
+	rule.Comment = raw.Comment
+
+	// iterate rule expressions looking for the single counter
+	for _, expr := range raw.Exprs {
+		hasKey := func(key string) bool { _, ok := expr[key]; return ok }
+		if hasKey("counter") {
+			rule.Counter = &Counter{}
+			if err := json.Unmarshal(expr["counter"], rule.Counter); err != nil {
+				return fmt.Errorf("unable to parse metadata: %w", err)
+			}
+			// we can return early since we are not looking for anything else
+			return nil
+		}
+	}
+	return nil
+}
+
+type Counter struct {
+	Packets int64 `json:"packets"`
+	Bytes   int64 `json:"bytes"`
+}
+
+func (*Nftables) SampleConfig() string {
+	return nftableConfig
+}
+
+func (nft *Nftables) Gather(acc telegraf.Accumulator) error {
+	if len(nft.Tables) == 0 {
+		return errors.New("invalid configuration - expected a `Tables` entry with list of nftables to monitor")
```

Comment:
Error message uses backticks inconsistently, Should use single quotes for consistency: 'Tables'